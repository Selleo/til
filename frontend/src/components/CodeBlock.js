import React from 'react'
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter'
import { darcula } from 'react-syntax-highlighter/dist/cjs/styles/prism'

class CodeBlock extends React.PureComponent {
  render() {
    const { language, value } = this.props

    return (
      <SyntaxHighlighter style={darcula} language={language}>
        {value}
      </SyntaxHighlighter>
    )
  }
}

export default CodeBlock
