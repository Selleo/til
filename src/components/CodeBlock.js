import React from "react";

import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { atomDark } from "react-syntax-highlighter/dist/cjs/styles/prism";

class CodeBlock extends React.PureComponent {
  render() {
    const { language, value } = this.props;

    return (
      <SyntaxHighlighter style={atomDark} language={language}>
        {value}
      </SyntaxHighlighter>
    );
  }
}

export default CodeBlock;
