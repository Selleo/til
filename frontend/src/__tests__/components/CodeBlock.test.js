import React from "react";
import { render } from "@testing-library/react";
import CodeBlock from "../../components/CodeBlock";

describe("<CodeBlock/>", () => {
  it("renders properly", () => {
    const { asFragment } = render(<CodeBlock />);

    expect(asFragment()).toMatchSnapshot();
  });
});
