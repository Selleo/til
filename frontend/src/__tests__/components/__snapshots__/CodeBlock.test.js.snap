// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock should render SyntaxHighlighter 1`] = `
<div>
  <div
    class="language-ts"
    style="color: rgb(169, 183, 198); font-family: Consolas, Monaco, 'Andale Mono', monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; background: rgb(43, 43, 43);"
  >
    <code
      class="language-ts"
      style="color: rgb(169, 183, 198); font-family: Consolas, Monaco, 'Andale Mono', monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
    >
      <span
        class="token console class-name"
      >
        console
      </span>
      <span
        class="token"
        style="color: rgb(169, 183, 198);"
      >
        .
      </span>
      <span
        class="token method property-access"
        style="color: rgb(255, 198, 109);"
      >
        log
      </span>
      <span
        class="token"
        style="color: rgb(169, 183, 198);"
      >
        (
      </span>
      <span
        class="token"
        style="color: rgb(106, 135, 89);"
      >
        "test"
      </span>
      <span
        class="token"
        style="color: rgb(169, 183, 198);"
      >
        )
      </span>
    </code>
  </div>
</div>
`;
